storage/memory/peer_store.go

```diff
@@ -92,6 +92,14 @@ func (s *peerStore) shardIndex(infoHash bittorrent.InfoHash, p bittorrent.Peer)
 	return idx
 }
 
+func (s *peerStore) shardIndexByAddressType(infoHash bittorrent.InfoHash, v6 bool) uint32 {
```

Comment:
Oh there is definitely one too many plus signs in there, that's true...
And yeah, the modulo should be with `len/2`, whoops.